numpy/_core/src/umath/dispatching.c

```diff
@@ -769,11 +769,18 @@ promote_and_get_info_and_ufuncimpl(PyUFuncObject *ufunc,
      * 2. Check all registered loops/promoters to find the best match.
      * 3. Fall back to the legacy implementation if no match was found.
      */
-    PyObject *info = PyArrayIdentityHash_GetItem(ufunc->_dispatch_cache,
-                (PyObject **)op_dtypes);
-    if (info != NULL && PyObject_TypeCheck(
+    PyObject *info = NULL;
+    int result = PyArrayIdentityHash_GetItemRef(ufunc->_dispatch_cache,
+                                                (PyObject **)op_dtypes, &info);
+    if (result == -1) {
+        return NULL;
+    }
+    if (result == 1 && PyObject_TypeCheck(
             PyTuple_GET_ITEM(info, 1), &PyArrayMethod_Type)) {
         /* Found the ArrayMethod and NOT a promoter: return it */
+        // TODO properly manage refcounts, this is safe because
+        // PyArrayMethodObject instances are never decref'd otherwise
```

Comment:
This is OK, just to note that in theory NumPy could replace them, but in practice it cannot happen right now.  I.e. the TODO may be actually be a very very subtle bug.